Shell_Sort/shell_sort.c

```diff
@@ -0,0 +1,54 @@
+#include <stdio.h>
+void shell_sort(int array[], int size)
+{
+    int i, j, k, temp;
+
+    for (i = size / 2; i > 0; i = i / 2)
+    {
+        // Do a gapped insertion sort for this gap size(i). 
+        // The first gap elements a[0..i-1] are already in gapped order 
+        // keep adding one more element until the entire array is gap sorted  
+        for (j = i; j < size; j++)
+        {
+            // add a[j] to the elements that have been gap sorted 
+            // save a[j] in temp and make space at position j
+            
+            temp = arrary[j]; 
+            
+            // shift earlier gap-sorted elements up until the correct location 
+            // for a[j] is found 
+            for(k = j; k >=i && array[k-i]>temp ; k = k - i)
+            {
+                    //  put temp (the original a[j]) in its correct location 
+                    array[k]=temp;
+
+            }
+        }
+    }
+}
+
+int main()
+{
+    int array[30];
```

Comment:
Make the array dynamic 
Its size should depend on user input

Reply:
Ok will do

Reply:
Declared the array with user defined size.